tests/browser/test_observe.py

```
@@ -0,0 +1,288 @@
+    # Create a fresh Notte session for each page to avoid side-effects.
+    with notte.Session(
+        headless=True,
+        enable_perception=False,
+        viewport_width=VIEWPORT_WIDTH,
+        viewport_height=VIEWPORT_HEIGHT,
+    ) as session:
+
+        save_snapshot(save_dir=save_dir, session=session, url=url)
+
+
+def save_snapshot_trajectory(url: str, task: str) -> None:
```

Ruff (0.12.2)

Redefinition of unused `url` from line 11

(F811)